### PR TITLE
Manual sync 'release-next' with upstream/main

### PR DESCRIPTION
**Pull request by aliok**
Let's see if problem in https://github.com/openshift/knative-eventing/pull/1578 is fixed

### CODE REVIEW

**Comment by aliok** [comment]
> @aliok: all tests passed!

@openshift-bot You're crazy!

**Comment by aliok** [comment]
/hold

Do not merge. This is just to prove that this kind of change fixes the issue.
I will have a 2nd look and see if we need to change every test image out there.

**Comment by aliok** [comment]
/retest

Getting 500 error.

File: openshift/e2e-common.sh [hunk 1]
```diff
@@ -208,6 +208,10 @@ function run_e2e_rekt_tests(){
   oc -n knative-eventing set env deployment/mt-broker-controller BROKER_INJECTION_DEFAULT=true || return 1
   wait_until_pods_running $EVENTING_NAMESPACE || return 2
 
+  echo "Replacing knative-eventing-test-event-library"
+  sed -i -e "s|registry.ci.openshift.org/openshift/knative-.*:knative-eventing-test-event-library|${KNATIVE_EVENTING_TEST_EVENT_LIBRARY}|g" "$(dirname "$0")/../test/rekt/resources/eventlibrary/eventlibrary.yaml"
+  echo "Replaced: $(dirname "$0")/../test/rekt/resources/eventlibrary/eventlibrary.yaml"
+  cat "$(dirname "$0")/../test/rekt/resources/eventlibrary/eventlibrary.yaml"
```
Comment (matzew) [inline openshift/e2e-common.sh:214]: I think we should do that for all `rekt test` images

Comment (aliok) [reply openshift/e2e-common.sh:214]: Yep, I was checking if this works or not. I will do it for all test images

**Comment by aliok** [comment]
cc @matzew 

I will cherry pick the changes to main and other places

**Comment by aliok** [comment]
Ah crap, creating `kodata` dirs didn't work because there are some vendored stuff 🤦 

Gotta really go and modify bash scripting to check if `kodata` dir exists and modify the Dockerfile based on that...

**Comment by aliok** [comment]
/test 47-conformance-aws-ocp-47

```
failed to apply Terraform: failed to complete the change
``

**Comment by aliok** [comment]
/hold

Intellij messed up reverting symlinks :(

File: Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 CGO_ENABLED=0
 GOOS=linux
-CORE_IMAGES=$(shell find ./cmd -name main.go ! -path "./cmd/broker/*" ! -path "./cmd/mtbroker/*" | sed 's/main.go//') ./vendor/knative.dev/pkg/apiextensions/storageversion/cmd/migrate ./vendor/knative.dev/pkg/leaderelection/chaosduck
+CORE_IMAGES=$(shell find ./cmd -name main.go ! -path "./cmd/broker/*" ! -path "./cmd/mtbroker/*" | sed 's|/main.go||') ./vendor/knative.dev/pkg/apiextensions/storageversion/cmd/migrate ./vendor/knative.dev/pkg/leaderelection/chaosduck
```
Comment (aliok) [inline Makefile:5]: This change removes the trailing slashes.

File: openshift/ci-operator/generate-dockerfiles.sh [hunk 3]
```diff
@@ -7,8 +7,14 @@ function generate_dockefiles() {
...
   for img in $@; do
     local image_base=$(basename $img)
+    local kodata_path="$img/kodata"
     mkdir -p $target_dir/$image_base
-    bin=$image_base envsubst < openshift/ci-operator/Dockerfile.in > $target_dir/$image_base/Dockerfile
+    if [ -d "$kodata_path" ]
+    then
+        bin=$image_base kodata_path=$kodata_path envsubst < openshift/ci-operator/Dockerfile_with_kodata.in > $target_dir/$image_base/Dockerfile
+    else
+        bin=$image_base envsubst < openshift/ci-operator/Dockerfile.in > $target_dir/$image_base/Dockerfile
```
Comment (matzew) [inline openshift/ci-operator/generate-dockerfiles.sh:16]: nice

File: openshift/ci-operator/knative-images/apiserver_receive_adapter/Dockerfile [hunk 4]
```diff
@@ -2,5 +2,6 @@
 FROM openshift/origin-base
 USER 65532
 
+COPY ./cmd/apiserver_receive_adapter/kodata /var/run/ko
```
Comment (matzew) [inline openshift/ci-operator/knative-images/apiserver_receive_adapter/Dockerfile:5]: I guess.... this does not cause any harm....  so I am OK.... but it worked before, without

Comment (aliok) [reply openshift/ci-operator/knative-images/apiserver_receive_adapter/Dockerfile:5]: yeah, I could've deleted the kodata dir of it but let's keep it. there are only some symlinks in it to retrieve git commit etc.

**Comment by matzew** [comment]
I think this looks good.

There is no need to actually merge it, since we have `release-next` recreated every night

**Comment by aliok** [comment]
@matzew 
> I think this looks good.
> There is no need to actually merge it, since we have release-next recreated every night

You mean I should just do the same changes in main and that's it?

File: openshift/ci-operator/knative-images/apiserver_receive_adapter/Dockerfile (hunk 4 above)
Comment (matzew) [reply openshift/ci-operator/knative-images/apiserver_receive_adapter/Dockerfile:5]: SOUNDS GOOD TO ME (famous last words :rofl: )

**Comment by matzew** [comment]
> You mean I should just do the same changes in main and that's it?

If we merge this now -> it's gone tomorrow - yes, we need to stash the releavant bits (E.g. the scripts and generator) on `main`

File: ci [hunk 5]
```diff
@@ -0,0 +1 @@
+Tue Jan 18 19:34:21 EST 2022
```
Comment (matzew) [inline ci:1]: this for instance we do not need on `main` :-)

**Comment by aliok** [comment]
/close

Closing in favor of the PR in main branch: https://github.com/openshift/knative-eventing/pull/1584

**Pull request closed without merging**